### PR TITLE
[MAD-15485] Support for explicit streaming images regiistration and usage

### PR DESCRIPTION
**Pull request by Aleks-Starykh-TR**
### Ticket:
[MAD-15485](https://jira.carbonated.com:8443/browse/MAD-15485)

### What does this PR do?
* Patches `AssetCatalog::GetAssetIdByPath()` to set correct `streamingImageAssetId.m_subId == 1000` for a preprocessed streaming image  being registered.
* Patches `CSprite::LoadSprite() `/ `CSprite::LoadImage()` to use fully qualified paths with the `.streamingimage` extension for preprocessed streaming images.
* Minor: restores correct (original) 1st argument in the call to `bool textureExists = CheckIfFileExists(spritePath, cacheRelativePath);` in `CSprite::DoesSpriteTextureAssetExist()`;

### How was this PR tested?
Game run. Hex sprites are shown, no extra errors noted.

### CODE REVIEW

**Review by Twolewis — APPROVED** [review]

**Comment by Aleks-Starykh-TR** [comment]
Was checking @aefimov3 proposal, so temporary set **DO NOT MERGE** label.
Came to conclusion that `CSprite` patches are mandatory, and `AssetCatalog` patch is highly desirable (it can be bypassed by repeating `AssetCatalog::GetAssetIdByPath()` code in client code with applied patch (`generatedID.m_subId = 1000`).

File: Gems/LyShine/Code/Source/Sprite.cpp [hunk 1]
```diff
@@ -751,7 +758,7 @@ bool CSprite::DoesSpriteTextureAssetExist(const AZStd::string& pathname)
 
     // Check if the texture asset exists
     const AZStd::string cacheRelativePath = AZStd::string::format("%s.%s", pathname.c_str(), streamingImageExtension);
-    bool textureExists = CheckIfFileExists(/* sourceRelativePath = */ pathname, cacheRelativePath);
```
Comment (carbonated-niko) [inline Gems/LyShine/Code/Source/Sprite.cpp:754]: Is this a carb specific engine change?

Comment (Aleks-Starykh-TR) [reply Gems/LyShine/Code/Source/Sprite.cpp:754]: This was patch by @aefimov3 , he fixed obvious bug in 2nd argument,
and I decided that substituting original `spritePath` with `pathname` in the 1st argument was a mistake.

Comment (aefimov3) [reply Gems/LyShine/Code/Source/Sprite.cpp:754]: I have a code piece that works locally that loads the sprites without engine patches. I sent you the code, please test it. It uses original lua scripts.

**Review by aefimov3 — CHANGES_REQUESTED** [review]
Please test the new code piece, if it works then we do not need this PR.

**Comment by JasonCarbonated** [comment]
Do we need this PR? What is the result of the code piece test @aefimov3 suggested a if working the changes  in this PR are uneeded?

**Comment by Aleks-Starykh-TR** [comment]
PR not needed after refactoring game code (registering new assets in game code).
Closed.

**Pull request closed without merging**